cl/opinion_page/tests.py

```diff
@@ -1240,6 +1318,64 @@ def test_form_save(self, mock) -> None:
         form.fields["second_judge"].queryset = qs
         form.fields["third_judge"].queryset = qs
 
+        self.assertEqual(form.is_valid(), True, msg=form.errors)
+
+        if form.is_valid():
+            form.save()
+
+        self.assertEqual(pre_count + 1, Opinion.objects.all().count())
+
+    def test_me_form_save(self, mock) -> None:
+        """Can we save maine form successfully to db?"""
+
+        pre_count = Opinion.objects.all().count()
+
+        form = MeCourtUploadForm(
+            self.me_data,
+            pk="me",
+            files={"pdf_upload": self.pdf},
+        )
+
+        print("panel", form.fields["panel"].queryset)
+
+        self.assertEqual(form.is_valid(), True, msg=form.errors)
+
+        if form.is_valid():
+            form.save()
+
+        self.assertEqual(pre_count + 1, Opinion.objects.all().count())
+
+    def test_mo_form_save(self, mock) -> None:
+        """Can we save missouri form successfully to db?"""
+
+        pre_count = Opinion.objects.all().count()
+
+        form = MoCourtUploadForm(
+            self.mo_data,
+            pk="mo",
+            files={"pdf_upload": self.pdf},
+        )
+
+        self.assertEqual(form.is_valid(), True, msg=form.errors)
+
+        if form.is_valid():
+            form.save()
+
+        self.assertEqual(pre_count + 1, Opinion.objects.all().count())
```

Comment:
I think this test should measure if the opinion was successfully added to the right court - 

    def test_mo_form_save(self, mock) -> None:
        """Can we save missouri form successfully to db?"""

        pre_count = Opinion.objects.filter(cluster__docket__court__id="mo").count()
        form = MoCourtUploadForm(
            self.mo_data,
            pk="mo",
            files={"pdf_upload": self.pdf},
        )
        self.assertEqual(form.is_valid(), True, msg=form.errors)
        if form.is_valid():
            form.save()
        post_save_count = Opinion.objects.filter(cluster__docket__court__id="mo").count()
        self.assertEqual(pre_count + 1, post_save_count)

something more akin to this ... where we check that the opinion was added to the right court  no?

Reply:
yes,  this is a better approach, we validate that the opinion was added to the correct court, not just validate that we created an opinion